conda_vendor/conda_vendor.py

```diff
@@ -658,13 +688,67 @@ def ironbank_gen(file: str, solver: str, platform: str):
     platform: str
         platform to vendor, e.g. linux-32, linux-64, osx-64, win-32
 
+    virtual_package_spec: str
+        location of virtual-packages.yml to be used by conda-lock.  The format
+        of the virtual-packages.yml is specified in the conda-lock
+        documentation.
+        Note: if a file named virutal-packages.yml exist in the directory
+        where conda-lock is run it will use that file.
+        see (https://github.com/conda-incubator/conda-lock).
     """
-    package_list = solve_environment(Path(file), solver, platform)
+    package_list = solve_environment(
+        Path(file), solver, platform, virtual_package_spec
+    )
     yaml_dump_ironbank_manifest(package_list)
 
 
+###########################################################################
+#                                                                         #
+#              conda-vendor virtual-packages                              #
+#                                                                         #
+###########################################################################
+@click.command("virtual-packages", help="dump host virtual packages")
+@click.option(
+    "--solver",
+    default="conda",
+    help="Solver to use. conda, mamba, micromamba",
+)
+@click.option(
+    "-o",
+    "--output",
+    default=None,
+    help="name of file to write.  Default is stdout.",
+)
+def virtual_packages(solver, output):
+    from subprocess import check_output
```

Comment:
Why import inside the function and not globally with other imports?

Reply:
because i only need it when this subcommand runs and i done't want to pull it in otherwise.  it's a choice.

What I'd really like to do is make each subcommand its own module, but that seems like overkill for this small little project.

Reply:
I'm not super concerned about it. Seems like a minor design choice.